controllers/routingctrl/reconcile_resources.go

```diff
@@ -17,11 +18,13 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func (r *Controller) reconcileResources(ctx context.Context, target *unstructured.Unstructured) error {
-	// TODO shouldn't we make it a predicate for ctrl watch instead
-	_, exportModeFound := extractExportModes(target)
-	if !exportModeFound {
-		return nil
+func (r *Controller) createRoutingResources(ctx context.Context, target *unstructured.Unstructured) error {
+	exportModes := extractExportModes(target, r.log)
+
+	if len(exportModes) == 0 {
+		r.log.Info("No export mode found for target")
+
+		return propagateHostsToWatchedCR(target, spi.RoutingTemplateData{}, r.log)
```

Comment:
What does passing `spi.RoutingTemplateData{}` mean? If it's only to trigger the logic of removal not needed annotations I think we should organize this functionality better.

I was thinking of simply removing relevant annotations here instead of calling with incomplete struct to only achieve the same effect:

```golang
metadata.ApplyMetaOptions(target,
	annotations.Remove(annotations.RoutingAddressesExternal("")),
	annotations.Remove(annotations.RoutingAddressesPublic("")),
)
```

where `Remove` is a new function:

```golang
func Remove(annotation Annotation) metadata.OptionFunc {
	return func(obj metav1.Object) {
		existingAnnotations := obj.GetAnnotations()
		if existingAnnotations == nil {
			existingAnnotations = make(map[string]string)
		}

		delete(existingAnnotations, annotation.Key())

		obj.SetAnnotations(existingAnnotations)
	}
}
```
and the last bit:

```golang
type OptionFunc func(obj metav1.Object)

func (f OptionFunc) ApplyToMeta(obj metav1.Object) {
	f(obj)
}
```

Reply:
Awesome, added `pkg/metadata/annotations/meta.go` with this to match labels structure

[006ede1](https://github.com/maistra/odh-platform/pull/48/commits/006ede174dfe8cc7ef5f734525ed32153d5e9656)